### PR TITLE
fix(notification-service): fix for notification user entry

### PR DESCRIPTION
**Pull request by mayank-SFIN571**
RFIT-568

## Description

Notification user creation issue resolved .

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: services/notification-service/src/controllers/notification.controller.ts [hunk 1]
```diff
@@ -67,14 +67,18 @@ export class NotificationController {
...
-
-    const notifUser = new NotificationUser();
     if (!notif || !notif.id) {
       throw new HttpErrors.UnprocessableEntity(AuthErrorKeys.UnknownError);
     }
-    notifUser.notificationId = notif.id;
-    notifUser.isRead = false;
-    await this.notificationUserRepository.create(notifUser);
+
+    const receiversToCreate = notif.receiver.to.map(to => {
```
Comment (samarpan-b) [inline services/notification-service/src/controllers/notification.controller.ts:74]: check for receiver null

File: services/notification-service/src/controllers/notification.controller.ts [hunk 2]
```diff
@@ -67,14 +67,18 @@ export class NotificationController {
...
+      const notifUser = new NotificationUser();
+      notifUser.notificationId = notif.id || '';
```
Comment (samarpan-b) [inline services/notification-service/src/controllers/notification.controller.ts:76]: use ??

File: services/notification-service/src/controllers/notification.controller.ts [hunk 3]
```diff
@@ -67,14 +67,18 @@ export class NotificationController {
...
+      notifUser.userId = to.id;
+      notifUser.isRead = false;
```
Comment (samarpan-b) [inline services/notification-service/src/controllers/notification.controller.ts:78]: no need to set. there is default

**Review by samarpan-b — CHANGES_REQUESTED** [review]

File: services/notification-service/src/controllers/notification.controller.ts (hunk 2 above)
Comment (mayank-SFIN571) [reply services/notification-service/src/controllers/notification.controller.ts:76]: to bypass type check failure

File: services/notification-service/src/controllers/notification.controller.ts (hunk 3 above)
Comment (mayank-SFIN571) [reply services/notification-service/src/controllers/notification.controller.ts:78]: sure

File: services/notification-service/src/controllers/notification.controller.ts [hunk 4]
```diff
@@ -67,14 +68,16 @@ export class NotificationController {
...
-
-    const notifUser = new NotificationUser();
     if (!notif || !notif.id) {
       throw new HttpErrors.UnprocessableEntity(AuthErrorKeys.UnknownError);
     }
-    notifUser.notificationId = notif.id;
-    notifUser.isRead = false;
-    await this.notificationUserRepository.create(notifUser);
+
+    if (!notif.receiver || !notif.receiver.to) {
```
Comment (samarpan-b) [inline services/notification-service/src/controllers/notification.controller.ts:75]: move this null check to inside the new function

**Review by samarpan-b — CHANGES_REQUESTED** [review]

**Review by samarpan-b — APPROVED** [review]

**Pull request merged**